tensorboard/defs/defs.bzl

```diff
@@ -39,61 +35,50 @@ def tf_js_binary(
     Args:
         name: Name of the target.
         compile: whether to compile when bundling. Only used internally.
-        deps: dependencies of the js_binary.
         visibility: visibility of the target.
         dev_mode_only: whether the binary is for development. When True, it will
-          omit the Terser.
+          omit the minification step.
         includes_polymer: whether this binary contains Polymer. Only used
           internally.
-        **kwargs: keyword arguments to rollup_bundle. Please refer to
-          https://bazelbuild.github.io/rules_nodejs/Built-ins.html#rollup_bundle
+        **kwargs: Other keyword arguments to esbuild(). Typically used for
+          entry_point and deps. Please refer to https://esbuild.github.io/api/
           for more details.
     """
 
-    # `compile` option is used internally but is not used by rollup_bundle.
-    # Discard it.
-    internal_rollup_name = name + "_rollup_internal_dbg"
-    rollup_bundle(
-        name = internal_rollup_name,
-        config_file = "//tensorboard/defs:rollup_config.js",
-        # Must pass `true` here specifically, else the input file argument to
-        # Rollup (appended by `rollup_binary`) is interpreted as a value for
-        # the preceding option.
-        args = ["--failAfterWarnings", "true", "--silent", "true"],
-        deps = deps + [
-            "@npm//@rollup/plugin-commonjs",
-            "@npm//@rollup/plugin-node-resolve",
-        ],
-        format = "iife",
-        sourcemap = "false",
-        visibility = ["//visibility:private"],
-        **kwargs
-    )
-
-    if dev_mode_only:
-        internal_result_name = internal_rollup_name
-    else:
-        internal_result_name = name + "_terser_internal_min"
-        terser_minified(
-            name = internal_result_name,
-            src = internal_rollup_name,
-            # Notes about the terser config:
-            # compress.passes - this is set to '1' to workaround issue with
-            #   terser and threejs. In practice it (surprisingly) generates
-            #   smaller results than when it was previously set to '3'.
-            config_file = "//tensorboard/defs:terser_config.json",
-            visibility = ["//visibility:private"],
-            sourcemap = False,
-        )
-
-    # For some reason, terser_minified is not visible from other targets. Copy
-    # or re-export seems to work okay.
-    native.genrule(
+    # esbuild is a fast JavaScript bundler[1] appropriate for both production
+    # and development builds.
+    #
+    # Bazel documents[2] how to use esbuild bundling with ts_project but we use
+    # the not-quite-deprecated ts_library rule instead of ts_project. We've
+    # managed to get esbuild working with ts_library but its long-term support
```

Comment:
Did you have to do anything besides specifying .mjs extensions to get ts_library to work?

Reply:
That's all I was certain was ts_library-specific.